src/Modules/AuthModule/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,6 +1,251 @@
+import { useForm } from "react-hook-form";
+import logo from "../../../../assets/images/PMS 3.svg";
+import { FormDataChangPass } from "../../../../interfaces/Auth";
+import "../Login/Login.css";
+import { useContext, useState } from "react";
+import { toast } from "react-toastify";
+import axios from "axios";
+import { AuthContext } from "../../../../context/AuthContext";
 
 export default function ChangePassword() {
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm<FormDataChangPass>();
+
+  const { baseUrl } = useContext(AuthContext);
+
+  // Password eye
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  const togglePasswordVisibility = (): void => {
```

Comment:
Instead of having two functions that will perform the same functionality, we can do the following:

 const togglePasswordVisibility = (setterFn)=> {

  return  ()=> setterFn((prevState: boolean) => !prevState);
  };

and in the button 

```jsx
//1st button
<button onClick={togglePasswordVisibility(setShowPassword)}
//2nd button
<button onClick={togglePasswordVisibility(setShowNewPassword)}
```

Reply:
and a better approach than that is to make PasswordInput component that will handle the password visibility inside it.  This will improve our code as we don't need to create 10 states if we have 10 inputs with type password.